lib/ansible/modules/network/aos/aos_blueprint.py

```diff
@@ -0,0 +1,291 @@
+#!/usr/bin/python
+#
+# (c) 2017 Apstra Inc, <community@apstra.com>
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+ANSIBLE_METADATA = {'status': ['preview'],
+                    'supported_by': 'community',
+                    'version': '1.0'}
+
+DOCUMENTATION = '''
+---
+module: aos_blueprint
+author: jeremy@apstra.com (@jeremyschulman)
+version_added: "2.3"
+short_description: Manage AOS blueprint instance
+description:
+    - Apstra AOS Blueprint module let you manage your Blueprint easily. You can create
+      create and delete Blueprint by Name or ID. You can also use it to retrieve
+      all data from a blueprint. This module is idempotent
+      and support the I(check) mode. It's using the AOS REST API.
+requirements:
+  - "aos-pyez >= 0.6.0"
+options:
+  session:
+    description:
+      - An existing AOS session as obtained by aos_login module
+    required: true
+  name:
+    description:
+      - Name of the Blueprint to manage.
+        Only one of I(name) or I(id) can be set.
+  id:
+    description:
+      - AOS Id of the IP Pool to manage (can't be used to create a new IP Pool),
+        Only one of I(name) or I(id)  can be set.
+  state:
+    description:
+      - Indicate what is the expected state of the Blueprint.
+    choices: ['present', 'absent', 'build-ready']
+    default: present
+  timeout:
+    description:
+      - When state='build-ready', this timeout identifies timeout in seconds to wait before
+        declaring a failure
+  template:
+    description:
+      - When creating a blueprint, this value identifies, by user-name, an existing engineering
```

Comment:
I assume to users of this module "by user-name" will make sense, or should be be "by name"?